src/components/Chart.tsx

```diff
@@ -0,0 +1,62 @@
+import { useEffect, useState } from "react";
+import GenerationTypeChart from "./GenerationType";
+import React from "react";
+
+export enum DataType {
+    generationType = "https://data.dev.elexon.co.uk/bmrs/api/v1/generation/outturn/summary?from=2023-07-12&to=2023-07-12"
+}
+
+export enum ChartStyle {
+    pie,
+    bar
+}
+
+export class SingleDimDataPoint {

```

Comment:
I think we should default to interfaces unless we need class functionality, see here: https://stackoverflow.com/questions/51716808/when-use-a-interface-or-class-in-typescript